Sources/StreamChatUI/Utils/ImageMerger.swift

```diff
@@ -0,0 +1,91 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import UIKit
+
+/// The orientation to be used for merge the images
+public enum ImageMergeOrientation {
+    /// Merge the given images in horizontal orientation.
+    /// The width of the resulting images will be the addition of the widths of all the images,
+    /// whereas the height of the resulting image will be equal to the max of heights in the images
+    case horizontal
+    
+    /// Merge the given images in vertical orientation.
+    /// The width of the resulting images will be equal to the max of widths in the images,
+    /// whereas the height of the resulting image will be the addition of the heights of all the images
+    case vertical
+}
+
+public protocol ImageMerging {
+    /// Merges the images provided in the array
+    /// - Parameters:
+    ///   - images: The images to combine
+    ///   - orientation: The orientation to be used for combining the images
+    /// - Returns: A combined image
+    func merge(
+        images: [UIImage],
+        orientation: ImageMergeOrientation
+    ) -> UIImage?
+}
+
+open class DefaultImageMerger: ImageMerging {
+    // Initializer required for subclasses
```

Comment:
we can double-mark this with `Note`: or `NOTE:`

@nuno-vieira your opinion here?